File: Beginner/RemoveDuplicatesFromSortedArray/RemoveDuplicateFromSortedArray.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 1

        if len(nums) == 0:
            return 0
        
        for r in range(1, len(nums)):
            if nums[r] != nums[r-1]:
                nums[l] = nums[r]
                l += 1
        return l 


s = Solution()
print("The new list is: ", s.removeDuplicates([1, 1, 2]))
print("The new list is: ", s.removeDuplicates([0,0,1,1,1,2,2,3,3,4]))

File: Beginner/RemoveDuplicatesFromSortedArray/RemoveDuplicateFromSortedArray.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
    public: 
        int removeDuplicates(std::vector<int>& nums) {
            int l = 0;

            for (const int num: nums) {
                if (l < 1 || num > nums[l - 1])
                    nums[l++] = num;
            }
            return l;
        }
};

int main() {
    Solution s;
    std::vector<int> nums1 = {1, 1, 2};
    std::vector<int> nums2 = {0,0,1,1,1,2,2,3,3,4};            
    std::cout << s.removeDuplicates(nums1) << std::endl;
    std::cout << s.removeDuplicates(nums2) << std::endl;

    return 0;
}
